### ISSUE TITLE
[Bug]: RetryError: RetryError[<Future at 0x7f9aaae9e830 state=finished raised APIConnectionError>]

### ISSUE THREAD

**Issue by Ma-Fukudama** [run-llama/llama_index#7479]
### Bug Description

午前中は動いていたソースを午後に動かすとAPIコネクションエラーが発生して実行できなくなります。

### Version

0.8.12

### Steps to Reproduce

午前中に実行していたソースを全く変えずに実行しました。

streamlitを利用しています。

### Relevant Logs/Tracbacks

```shell
Traceback (most recent call last):
  File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connection.py", line 174, in _new_conn
    conn = connection.create_connection(
  File "/home/heart/.local/lib/python3.10/site-packages/urllib3/util/connection.py", line 72, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/lib/python3.10/socket.py", line 955, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 714, in urlopen
    httplib_response = self._make_request(
  File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 403, in _make_request
    self._validate_conn(conn)
  File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 1053, in _validate_conn
    conn.connect()
  File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connection.py", line 363, in connect
    self.sock = conn = self._new_conn()
  File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connection.py", line 186, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f9aaae9eb60>: Failed to establish a new connection: [Errno -2] Name or service not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/heart/.local/lib/python3.10/site-packages/requests/adapters.py", line 486, in send
    resp = conn.urlopen(
  File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 826, in urlopen
    return self.urlopen(
  File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 826, in urlopen
    return self.urlopen(
  File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 798, in urlopen
    retries = retries.increment(
  File "/home/heart/.local/lib/python3.10/site-packages/urllib3/util/retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/embeddings (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f9aaae9eb60>: Failed to establish a new connection: [Errno -2] Name or service not known'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/heart/.local/lib/python3.10/site-packages/openai/api_requestor.py", line 520, in request_raw
    result = _thread_context.session.request(
  File "/home/heart/.local/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/heart/.local/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/home/heart/.local/lib/python3.10/site-packages/requests/adapters.py", line 519, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/embeddings (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f9aaae9eb60>: Failed to establish a new connection: [Errno -2] Name or service not known'))

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/heart/.local/lib/python3.10/site-packages/tenacity/__init__.py", line 382, in __call__
    result = fn(*args, **kwargs)
  File "/home/heart/.local/lib/python3.10/site-packages/llama_index/embeddings/openai.py", line 122, in get_embedding
    return openai.Embedding.create(input=[text], model=engine, **kwargs)["data"][0][
  File "/home/heart/.local/lib/python3.10/site-packages/openai/api_resources/embedding.py", line 33, in create
    response = super().create(*args, **kwargs)
  File "/home/heart/.local/lib/python3.10/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 153, in create
    response, _, api_key = requestor.request(
  File "/home/heart/.local/lib/python3.10/site-packages/openai/api_requestor.py", line 220, in request
    result = self.request_raw(
  File "/home/heart/.local/lib/python3.10/site-packages/openai/api_requestor.py", line 533, in request_raw
    raise error.APIConnectionError(
openai.error.APIConnectionError: Error communicating with OpenAI: HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/embeddings (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f9aaae9eb60>: Failed to establish a new connection: [Errno -2] Name or service not known'))

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/heart/.local/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 552, in _run_script
    exec(code, module.__dict__)
  File "/home/heart/llama_index/examples/paul_graham_essay/screen0823.py", line 93, in <module>
    main()
  File "/home/heart/llama_index/examples/paul_graham_essay/screen0823.py", line 80, in main
    response = response_generator.generate(question)
  File "/home/heart/llama_index/examples/paul_graham_essay/screen0823.py", line 46, in generate
    result = engine.query(question+"回答には出典のリンクを含めるようにしてください。")
  File "/home/heart/.local/lib/python3.10/site-packages/llama_index/indices/query/base.py", line 23, in query
    response = self._query(str_or_query_bundle)
  File "/home/heart/.local/lib/python3.10/site-packages/llama_index/query_engine/retriever_query_engine.py", line 165, in _query
    nodes = self.retrieve(query_bundle)
  File "/home/heart/.local/lib/python3.10/site-packages/llama_index/query_engine/retriever_query_engine.py", line 113, in retrieve
    nodes = self._retriever.retrieve(query_bundle)
  File "/home/heart/.local/lib/python3.10/site-packages/llama_index/indices/base_retriever.py", line 22, in retrieve
    return self._retrieve(str_or_query_bundle)
  File "/home/heart/.local/lib/python3.10/site-packages/llama_index/indices/list/retrievers.py", line 75, in _retrieve
    query_embedding, node_embeddings = self._get_embeddings(query_bundle, nodes)
  File "/home/heart/.local/lib/python3.10/site-packages/llama_index/indices/list/retrievers.py", line 101, in _get_embeddings
    self._index._service_context.embed_model.get_agg_embedding_from_queries(
  File "/home/heart/.local/lib/python3.10/site-packages/llama_index/embeddings/base.py", line 123, in get_agg_embedding_from_queries
    query_embeddings = [self.get_query_embedding(query) for query in queries]
  File "/home/heart/.local/lib/python3.10/site-packages/llama_index/embeddings/base.py", line 123, in <listcomp>
    query_embeddings = [self.get_query_embedding(query) for query in queries]
  File "/home/heart/.local/lib/python3.10/site-packages/llama_index/embeddings/base.py", line 94, in get_query_embedding
    query_embedding = self._get_query_embedding(query)
  File "/home/heart/.local/lib/python3.10/site-packages/llama_index/embeddings/openai.py", line 282, in _get_query_embedding
    return get_embedding(
  File "/home/heart/.local/lib/python3.10/site-packages/tenacity/__init__.py", line 289, in wrapped_f
    return self(f, *args, **kw)
  File "/home/heart/.local/lib/python3.10/site-packages/tenacity/__init__.py", line 379, in __call__
    do = self.iter(retry_state=retry_state)
  File "/home/heart/.local/lib/python3.10/site-packages/tenacity/__init__.py", line 326, in iter
    raise retry_exc from fut.exception()
tenacity.RetryError: RetryError[<Future at 0x7f9aaae9e830 state=finished raised APIConnectionError>]
```

**Comment by logan-markewich** [comment]
@Ma-Fukudama Try setting the openai key on the imported module directly, in addition to the environment variable.

This error usually happens in a notebook.

```
import os
import openai

os.environ["OPENAI_API_KEY"] = "sk-..."
opanai.api_key = os.environ["OPENAI_API_KEY"] 
```

**Comment by Ma-Fukudama** [comment]
@logan-markewich 
I tried setting OPENAI_API_KEY in the source, but no change.
The incident occurred today around 17:00 Japan time.
It was working fine in the morning.

**Comment by wilberlee** [comment]
> ### 错误描述
> 如果你在下午运行一个上午运行过的源，将会出现 API 连接错误，并且你将无法运行它。
> 
> ### 版本
> 0.8.12
> 
> ### 重现步骤
> 我运行了早上运行的源代码，根本没有改变它。
> 
> 我正在使用streamlit。
> 
> ### 相关日志/回溯
> ```shell
> Traceback (most recent call last):
>   File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connection.py", line 174, in _new_conn
>     conn = connection.create_connection(
>   File "/home/heart/.local/lib/python3.10/site-packages/urllib3/util/connection.py", line 72, in create_connection
>     for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
>   File "/usr/lib/python3.10/socket.py", line 955, in getaddrinfo
>     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
> socket.gaierror: [Errno -2] Name or service not known
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 714, in urlopen
>     httplib_response = self._make_request(
>   File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 403, in _make_request
>     self._validate_conn(conn)
>   File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 1053, in _validate_conn
>     conn.connect()
>   File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connection.py", line 363, in connect
>     self.sock = conn = self._new_conn()
>   File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connection.py", line 186, in _new_conn
>     raise NewConnectionError(
> urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f9aaae9eb60>: Failed to establish a new connection: [Errno -2] Name or service not known
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/home/heart/.local/lib/python3.10/site-packages/requests/adapters.py", line 486, in send
>     resp = conn.urlopen(
>   File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 826, in urlopen
>     return self.urlopen(
>   File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 826, in urlopen
>     return self.urlopen(
>   File "/home/heart/.local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 798, in urlopen
>     retries = retries.increment(
>   File "/home/heart/.local/lib/python3.10/site-packages/urllib3/util/retry.py", line 592, in increment
>     raise MaxRetryError(_pool, url, error or ResponseError(cause))
> urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/embeddings (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f9aaae9eb60>: Failed to establish a new connection: [Errno -2] Name or service not known'))
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/home/heart/.local/lib/python3.10/site-packages/openai/api_requestor.py", line 520, in request_raw
>     result = _thread_context.session.request(
>   File "/home/heart/.local/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
>     resp = self.send(prep, **send_kwargs)
>   File "/home/heart/.local/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
>     r = adapter.send(request, **kwargs)
>   File "/home/heart/.local/lib/python3.10/site-packages/requests/adapters.py", line 519, in send
>     raise ConnectionError(e, request=request)
> requests.exceptions.ConnectionError: HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/embeddings (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f9aaae9eb60>: Failed to establish a new connection: [Errno -2] Name or service not known'))
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/home/heart/.local/lib/python3.10/site-packages/tenacity/__init__.py", line 382, in __call__
>     result = fn(*args, **kwargs)
>   File "/home/heart/.local/lib/python3.10/site-packages/llama_index/embeddings/openai.py", line 122, in get_embedding
>     return openai.Embedding.create(input=[text], model=engine, **kwargs)["data"][0][
>   File "/home/heart/.local/lib/python3.10/site-packages/openai/api_resources/embedding.py", line 33, in create
>     response = super().create(*args, **kwargs)
>   File "/home/heart/.local/lib/python3.10/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 153, in create
>     response, _, api_key = requestor.request(
>   File "/home/heart/.local/lib/python3.10/site-packages/openai/api_requestor.py", line 220, in request
>     result = self.request_raw(
>   File "/home/heart/.local/lib/python3.10/site-packages/openai/api_requestor.py", line 533, in request_raw
>     raise error.APIConnectionError(
> openai.error.APIConnectionError: Error communicating with OpenAI: HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/embeddings (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f9aaae9eb60>: Failed to establish a new connection: [Errno -2] Name or service not known'))
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "/home/heart/.local/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 552, in _run_script
>     exec(code, module.__dict__)
>   File "/home/heart/llama_index/examples/paul_graham_essay/screen0823.py", line 93, in <module>
>     main()
>   File "/home/heart/llama_index/examples/paul_graham_essay/screen0823.py", line 80, in main
>     response = response_generator.generate(question)
>   File "/home/heart/llama_index/examples/paul_graham_essay/screen0823.py", line 46, in generate
>     result = engine.query(question+"回答には出典のリンクを含めるようにしてください。")
>   File "/home/heart/.local/lib/python3.10/site-packages/llama_index/indices/query/base.py", line 23, in query
>     response = self._query(str_or_query_bundle)
>   File "/home/heart/.local/lib/python3.10/site-packages/llama_index/query_engine/retriever_query_engine.py", line 165, in _query
>     nodes = self.retrieve(query_bundle)
>   File "/home/heart/.local/lib/python3.10/site-packages/llama_index/query_engine/retriever_query_engine.py", line 113, in retrieve
>     nodes = self._retriever.retrieve(query_bundle)
>   File "/home/heart/.local/lib/python3.10/site-packages/llama_index/indices/base_retriever.py", line 22, in retrieve
>     return self._retrieve(str_or_query_bundle)
>   File "/home/heart/.local/lib/python3.10/site-packages/llama_index/indices/list/retrievers.py", line 75, in _retrieve
>     query_embedding, node_embeddings = self._get_embeddings(query_bundle, nodes)
>   File "/home/heart/.local/lib/python3.10/site-packages/llama_index/indices/list/retrievers.py", line 101, in _get_embeddings
>     self._index._service_context.embed_model.get_agg_embedding_from_queries(
>   File "/home/heart/.local/lib/python3.10/site-packages/llama_index/embeddings/base.py", line 123, in get_agg_embedding_from_queries
>     query_embeddings = [self.get_query_embedding(query) for query in queries]
>   File "/home/heart/.local/lib/python3.10/site-packages/llama_index/embeddings/base.py", line 123, in <listcomp>
>     query_embeddings = [self.get_query_embedding(query) for query in queries]
>   File "/home/heart/.local/lib/python3.10/site-packages/llama_index/embeddings/base.py", line 94, in get_query_embedding
>     query_embedding = self._get_query_embedding(query)
>   File "/home/heart/.local/lib/python3.10/site-packages/llama_index/embeddings/openai.py", line 282, in _get_query_embedding
>     return get_embedding(
>   File "/home/heart/.local/lib/python3.10/site-packages/tenacity/__init__.py", line 289, in wrapped_f
>     return self(f, *args, **kw)
>   File "/home/heart/.local/lib/python3.10/site-packages/tenacity/__init__.py", line 379, in __call__
>     do = self.iter(retry_state=retry_state)
>   File "/home/heart/.local/lib/python3.10/site-packages/tenacity/__init__.py", line 326, in iter
>     raise retry_exc from fut.exception()
> tenacity.RetryError: RetryError[<Future at 0x7f9aaae9e830 state=finished raised APIConnectionError>]
> ```

Hello, I encountered the same error when deploying the project in an intranet environment. The intranet environment does not have access to the internet. I am loading a local model instead of using OpenAI's interface. I reproduced this issue on a computer in an internet-connected environment: I first connected to the internet and installed llama-index, then disconnected from the network. When I tried to import llama_index for the first time, I received the same error as mentioned above.

**Comment by wilberlee** [comment]
Hello, I encountered the same error when deploying the project in an intranet environment. The intranet environment does not have access to the internet. I am loading a local model instead of using OpenAI's interface. I reproduced this issue on a computer in an internet-connected environment: I first connected to the internet and installed llama-index, then disconnected from the network. When I tried to import llama_index for the first time, I received the same error as mentioned above.
